habitat-lab/habitat/tasks/rearrange/multi_task/pddl_sim_state.py

```diff
@@ -183,8 +183,11 @@ def set_state(
                 agent=agent,
             )
 
-            agent.base_pos = start_pos
-            agent.base_rot = start_rot
+            # We only teleport the agent when it is allowed
+            if sim_info.enable_teleport_agent_pos:
```

Comment:
@eundersander  Thank you for the comments. I think this should be fine here. Also, set_state also needs to set the grasping function, so it will not make it a no-op. But @ASzot could also commend on this, thank you!

Reply:
I agree with Eric that this is very risky and I don't like changing this part of the code that all other systems rely on. I also don't think this needs to be a special case. 

I don't understand why Spot does not want to have the teleport condition in the PDDL? If you don't want Spot to teleport for navigation then you should use the "oracle" nav instead of the "teleport" nav or set `apply_postconds: False` for the navigation skill.

Reply:
@ASzot Thanks for this. The reason for not doing teleporting for Spot is because that Spot would be too closed to the target receptacle and thus prone to have collision.

Example of this 
https://user-images.githubusercontent.com/55121504/236859124-a8fdbc45-ab09-40c1-bd1c-6892b7aeda76.mp4

You can see that Spot navigates to the target receptacle and then was teleported. 

Based on 60 testing episodes, I observed that by removing this post-condition, we can reduce the collision rate from 0.36% to 0.11%, which is over 2 times smaller.

I have been already using `oracle_nav` (`nav_to_obj`). But when I turn off `apply_postconds`, it seems that the robot is not navigated anymore. See the video below

https://user-images.githubusercontent.com/55121504/236860287-a40a6702-dd63-4856-b712-bf81fe006033.mp4

I will work this a bit, thanks!